src/connectors/inventory/impl.js

```diff
@@ -90,6 +91,57 @@ module.exports = new class extends Connector {
         return validate(transformed);
     }
 
+    async getSystemProfileBatch (ids = [], refresh = false, retries = 2) {
+        if (ids.length === 0) {
+            return {};
+        }
+
+        ids = _.sortBy(ids);
+
+        if (ids.length > pageSize) {
+            const chunks = _.chunk(ids, pageSize);
+            const results = await P.map(chunks, chunk => this.getSystemProfileBatch(chunk, refresh));
+            return _.assign({}, ...results);
+        }
+
+        const uri = this.buildHostsUri();
+        uri.segment(ids.join());
+        uri.segment('system_profile');
```

Comment:
Like with the xjoin connector you can now limit the fields to only those you're interested in (owner id) using sparse fieldset.

Reply:
ah!  that's great to know, I'll make that change.